src/controllers/admin.js

```diff
@@ -348,21 +348,50 @@ adminController.navigation.get = function(req, res, next) {
 };
 
 adminController.homepage.get = function(req, res, next) {
-	plugins.fireHook('filter:homepage.get', {routes: [
-		{
-			route: 'categories',
-			name: 'Categories'
+	async.waterfall([
+		function(next) {
+			db.getSortedSetRange('cid:0:children', 0, -1, next);
+		},
+		function(cids, next) {
+			privileges.categories.filterCids('find', cids, 0, next);
 		},
-		{
-			route: 'recent',
-			name: 'Recent'
+		function(cids, next) {
+			categories.getMultipleCategoryFields(cids, ['name', 'slug'], next);
 		},
-		{
-			route: 'popular',
-			name: 'Popular'
+		function(categoryData, next) {
+			async.map(categoryData, function(category, next) {
+				var route = 'category/' + category.slug;
+
+				next(null, {
+					route: route,
+					name: 'Category: ' + category.name
+				});
+			}, next);
```

Comment:
Can you rewrite the above to not use async, there is nothing async in that block.

```
categoryData = categoryData.map(function(category) {
    return {
        route: 'category/' + category.slug,
        name: 'Category: ' + category.name
    };
});
next(null, categoryData);
```